src/Datadog.Trace.ClrProfiler.Managed.Loader/Startup.cs

```diff
@@ -0,0 +1,35 @@
+using System;
+using System.Reflection;
+
+namespace Datadog.Trace.ClrProfiler.Managed.Loader
+{
+    /// <summary>
+    /// A class that attempts to load the Datadog.Trace.ClrProfiler.Managed .NET assembly.
+    /// </summary>
+    public class Startup
+    {
+        /// <summary>
+        /// Initializes static members of the <see cref="Startup"/> class.
+        /// This method also attempts to load the Datadog.Trace.ClrProfiler.Managed. NET assembly.
+        /// </summary>
+        static Startup()
+        {
+            LoadManagedAssembly();
+        }
+
+        internal static bool ManagedAssemblyFound { get; set; }
+
+        private static void LoadManagedAssembly()
+        {
+            try
+            {
+                Assembly.Load(new AssemblyName("Datadog.Trace.ClrProfiler.Managed, Version=1.6.0.0, Culture=neutral, PublicKeyToken=def86d061d0d2eeb"));
```

Comment:
We may want to tie this to some sort of constant if possible. Is the current idea to update this manually for every release? I'm okay with that for now.

Reply:
Oh yeah, that was an item I wanted to revisit. I hard-coded it to start but I would much prefer making it more automatic if we can figure out a good way to do that.

Reply:
At worst, tests will fail if we miss it currently.

Reply:
The scarier part is if we miss this on the version update...maybe we should also enable running the smoke tests for the packaging step?

Reply:
Won't the tests fail on the version update PR if we miss this piece?

Reply:
Good question. I'm running an integration test run with a mismatching assembly version right now (on some other test branch) to test this scenario.

Reply:
Looks like our integration tests don't catch this because the changes only apply to Windows and our smoke tests (`Category="Smoke"`) don't count spans. I'll try enabling one `Category="EndToEnd"` (I think the HttpMessageHandler) that doesn't have external dependencies

Reply:
Heh yeah I got the `Datadog.Trace.ClrProfiler.IntegrationTests.HttpClientTests.HttpClient` to pass with the correct assembly name but fail with the incorrect assembly name. I'll get this into this PR 👍

Reply:
Currently blocked on adding this until the following issue is fixed with the [.NET Core SDK installer task](https://developercommunity.visualstudio.com/content/problem/656401/net-core-sdkruntime-installer-build-task-fails-wit.html) ☹️

Reply:
Okay I just pushed 00c009093911681b0c02ddc2d01d5901c13d00d2 which runs the HttpClientTests E2E tests on Windows. I was able to show on a separate branch that this fails when the string in the Loader fails. That's because we expect at least one span in the integration tests and we don't end up sending any in this scenario

Reply:
I'll leave this comment chain as unresolved to indicate that there's still some work left here to do, but I don't think I'll be changing this on this PR, if that's fine with you guys.